responsibleai/tests/causal/test_causal_general.py

```diff
@@ -10,6 +10,7 @@
 
 def test_causal_classification_scikitlearn_issue():
     # This test gets stuck on SciKit-Learn v1.1.0
+    # Issue appears fixed with v1.1.1
```

Comment:
So... it seems the most reasonable fix would be to say >1.1 but that excludes lots of older versions that do work as well. Am I right in assuming we don't want to be that strict because lots of users are using these older versions?

Reply:
Double checks... I hadn't realised that `!=` was allowed:
https://peps.python.org/pep-0508/

Reply:
However, per my other comment, I'm rather worried that this will break backwards compatibility. I understand that SciKit-Learn doesn't tend to be pickle-compatible between versions?

Reply:
This would mean we have to be quite strict on versions, wouldn't it?

Reply:
Or, as we were just discussing in the other meeting, start thinking some more about MLFlow.

Reply:
Should this test just run for scikit-learn version greater than 1.0?

Reply:
>= 1.0 AND ~= 1.1.0 ?
Not sure how one does the AND logic here...

Reply:
Reviving this thread: This seems important in that we don't allow people to use the latest 3 versions of sklearn with the current logic in `main`. Any objections with merging as is until we want to require `>1.1.0` at which point this special case goes away anyway?

@imatiach-msft @gaugup
Not sure if @riedgar-ms still has thoughts on this, but if so please share them, too.